crates/goose-cli/src/recipes/recipe.rs

```diff
@@ -0,0 +1,468 @@
+use anyhow::Result;
+use console::style;
+
+use crate::recipes::search_recipe::retrieve_recipe_file;
+use goose::recipe::{Recipe, RecipeParameter, RecipeParameterRequirement};
+use minijinja::{Environment, Template, UndefinedBehavior};
+use serde_json::Value as JsonValue;
+use serde_yaml::Value as YamlValue;
+use std::collections::{HashMap, HashSet};
+
+/// Loads and validates a recipe from a YAML or JSON file
+///
+/// # Arguments
+///
+/// * `path` - Path to the recipe file (YAML or JSON)
+/// * `log`  - whether to log information about the recipe or not
+/// * `params` - optional parameters to render the recipe with
+///
+/// # Returns
+///
+/// The parsed recipe struct if successful
+///
+/// # Errors
+///
+/// Returns an error if:
+/// - The file doesn't exist
+/// - The file can't be read
+/// - The YAML/JSON is invalid
+/// - The required fields are missing
+pub fn load_recipe(
+    recipe_name: &str,
+    log: bool,
+    params: Option<Vec<(String, String)>>,
+) -> Result<Recipe> {
+    let recipe_file_content = retrieve_recipe_file(recipe_name)?;
+
+    let recipe_parameters = validate_recipe_file_parameters(&recipe_file_content)?;
+
+    let mut params_for_template: Option<HashMap<String, String>> = None;
+    let rendered_content = match params {
+        None => recipe_file_content,
+        Some(user_params) => {
+            params_for_template =
+                Some(apply_values_to_parameters(&user_params, recipe_parameters)?);
+            render_content_with_params(&recipe_file_content, params_for_template.as_ref().unwrap())?
+        }
+    };
```

Comment:
We can avoid the mutable hashmap + `unwrap()` later with an `if let Some(...)`
```suggestion
let (rendered_content, params_for_template) = if let Some(user_params) = params {
    let map = apply_values_to_parameters(&user_params, recipe_parameters)?;
    let rendered = render_content_with_params(&recipe_file_content, &map)?;
    (rendered, Some(map))
} else {
    (recipe_file_content, None)
};
```